raiden-ts/tests/unit/raiden.spec.ts

```diff
@@ -120,9 +121,79 @@ function makeDummyDependencies(): RaidenEpicDeps {
 }
 
 describe('Raiden', () => {
+  function initEpicMock(action$: Observable<RaidenAction>): Observable<raidenSynced> {
+    return action$.pipe(
+      filter(raidenStarted.is),
+      mapTo(
+        raidenSynced({
+          tookMs: 9,
+          initialBlock: 1,
+          currentBlock: 10,
+        }),
+      ),
+    );
+  }
+
   test('address', () => {
     const deps = makeDummyDependencies();
     const raiden = new Raiden(dummyState, deps, combineRaidenEpics(of(dummyEpic)), dummyReducer);
     expect(raiden.address).toBe(dummyState.address);
   });
+
+  test('start', async () => {
+    const deps = makeDummyDependencies();
+    const raiden = new Raiden(
+      dummyState,
+      deps,
+      combineRaidenEpics(of(initEpicMock)),
+      dummyReducer,
+    );
+    expect(raiden.network).toEqual({ name: 'test', chainId: 1337 });
+    expect(raiden.log).toEqual(expect.objectContaining({ name: `raiden:${raiden.address}` }));
+    await expect(raiden.start()).resolves.toBeUndefined();
+    expect(raiden.started).toEqual(true);
+  });
+
+  test('stop', async (done) => {
+    const deps = makeDummyDependencies();
+
+    const raiden = new Raiden(
+      dummyState,
+      deps,
+      combineRaidenEpics(of(initEpicMock)),
+      dummyReducer,
+    );
+    raiden.action$.pipe(last()).subscribe((action) => {
+      expect(action).toEqual(raidenShutdown(expect.anything()));
+      done();
+    });
+    await expect(raiden.start()).resolves.toBeUndefined();
+    expect(raiden.started).toEqual(true);
+    await raiden.stop();
```

Comment:
Please, don't mix `done` with `async` tests. `action$.toPromise()` will return a promise which resolves to the latest emitted value, so you should be able to do something like:

```suggestion
    await expect(raiden.start()).resolves.toBeUndefined();
    expect(raiden.started).toEqual(true);
    const lastPromise = raiden.action$.toPromise();
    await expect(raiden.stop()).resolves.toBeUndefined();
    await expect(lastPromise).resolves.toEqual(raidenShutdown({ reason: ShutdownReason.STOP }));
```

Reply:
ok will follow the promise pattern

Reply:
You should also check that `raiden.started` is `false` after stopping.

Reply:
Ok I tried to implement this but I always get `timeout` errors but with the `done` approach it works fine. So any pointers where it could be improved.
```
Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:
```

Reply:
ok I forgot to remove the passed `done` function, which I was not calling, after removing it,  its working now.